onnx/checker.cc

```diff
@@ -0,0 +1,247 @@
+#include "onnx/checker.h"
+
+#include "onnx/defs/schema.h"
+#include "onnx/utils.h"
+
+#include <unordered_set>
+
+namespace onnx {
+namespace checker {
+
+#define fail(...) throw ValidationError(onnx::MakeString(__VA_ARGS__));
+
+#define enforce_has_field(proto, field)                                    \
```

Comment:
wrap everything into `do {} while(0)` otherwise it'd misfire if used in an if statement

Reply:
will fix it